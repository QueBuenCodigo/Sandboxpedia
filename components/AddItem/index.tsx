import React from "react";
import styled from "styled-components";

interface Props{
  onClick?:(event: React.MouseEvent<HTMLButtonElement>)=>void;
}
const AddItem:React.FC<Props> = (props) => {
  const {onClick} = props;
  return (
    <StyledAddItem onClick={onClick}>
      <div>Add Item</div>
    </StyledAddItem>
  );
}; 

const StyledAddItem = styled.button`
  display: flex;
  height: 100%;
  width: 100%;
  cursor: pointer;
  outline: 0;
  box-sizing: border-box;
  align-items: center;
  justify-content: center;
  margin:10px 0;
  padding: 12px 30px;
  border: dashed 1px #4eb7f5;
  background-color: rgba(255, 255, 255, 0.48);
  border-radius: 8px;
  &:hover {
      background-color: #daeffe;
    }
  div {
    flex-direction: row;
    justify-content: center;
    align-items: center;
    color: #6db4f0;
   
  }
`;

export default AddItem;
