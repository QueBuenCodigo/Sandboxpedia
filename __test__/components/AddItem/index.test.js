import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import AddItem from '../../../components/AddItem/index';


describe('<AddItem/>',()=>{

    const additem=mount(<AddItem></AddItem>)

    test('test del renderizado de AddItem',()=>{

        expect(additem.length).toEqual(1);
    });
});



describe(' AddItem Snapshot',()=>{

    test('comprobar la UI del componente AddItem',()=>{
        const additem= create(<AddItem/>);
        expect(additem.toJSON()).toMatchSnapshot();
    });

});
