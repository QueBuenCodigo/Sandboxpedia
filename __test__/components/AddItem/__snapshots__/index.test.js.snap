// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` AddItem Snapshot comprobar la UI del componente AddItem 1`] = `
<button
  className="AddItem__StyledAddItem-sc-1ikvgje-0 jVchal"
>
  <div>
    Add Item
  </div>
</button>
`;
